src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -471,6 +471,26 @@ def delete_arc_agents(release_namespace, kube_config, kube_context, helm_client_
                                "Helm release deletion failed: " + error_helm_delete.decode("ascii") +
                                " Please run 'helm delete azure-arc' to ensure that the release is deleted.")
     ensure_namespace_cleanup()
+    # Cleanup azure-arc-release NS if present (created during helm installation)
+    cleanup_release_install_namespace_if_exists()
+
+
+def cleanup_release_install_namespace_if_exists():
+    api_instance = kube_client.CoreV1Api()
+    try:
+        api_instance.read_namespace(consts.Release_Install_Namespace)
+    except Exception as ex:
+        if ex.status == 404:
+            # Nothing to delete, exiting here
+            return
+        else:
+            kubernetes_exception_handler(ex, consts.Get_Kubernetes_Helm_Release_Namespace_Fault_Type, error_message='Unable to fetch details about existense of kubernetes namespace: {}'.format(consts.Release_Install_Namespace), summary='Unable to fetch kubernetes namespace: {}'.format(consts.Release_Install_Namespace))
+
+    # If namespace exists, delete it
```

Comment:
should we even attempt to delete the namespace if we encountered an error while reading it in the above except -> else case?